Riot/Modules/AuthenticatedSession/AuthenticatedSessionViewControllerFactory.swift

```diff
@@ -0,0 +1,223 @@
+            guard let self = self else {
+                return
+            }
+            
+            guard let authenticationSession = authenticationSession, let flows = authenticationSession.flows else {
+                onFailure(AuthenticatedSessionViewControllerFactoryError.flowNotSupported)
+                return
+            }
+            
+            // Return the corresponding VC
+            if self.hasPasswordFlow(inFlows: flows) {
+                let authViewController = self.createPasswordViewController(title: title,
+                                                                           message: message,
+                                                                           authenticationSession: authenticationSession,
+                                                                           onAuthenticated: onAuthenticated,
+                                                                           onCancelled: onCancelled,
+                                                                           onFailure: onFailure)
+                onViewController(authViewController)
+            } else {
+                // Flow not supported yet
+                onFailure(AuthenticatedSessionViewControllerFactoryError.flowNotSupported)
+            }
+            
+        }, failure: { (error) in
+            guard let error = error as NSError? else {
+                return
+            }
+            
+            onFailure(error)
+        })
+    }
+    
+    /// Check if we support the authentication flow for a given request.
+    ///
+    /// - Parameters:
+    ///   - path: the request path.
+    ///   - httpMethod: the request http method.
+    ///   - onCancelled: the block called when the user cancelled the authentication.
+    ///   - onFailure: the blocked called on error.
+    func hasSupport(forPath path: String,
+                    httpMethod: String,
+                    success: @escaping (Bool) -> Void,
+                    failure: @escaping (NSError) -> Void) -> MXHTTPOperation {
+        
+        // Get the authentication flow required for this API
+        return session.matrixRestClient.authSessionForRequest(withMethod: httpMethod, path: path, parameters: [:], success: { [weak self] (authenticationSession) in
+            guard let self = self else {
+                return
+            }
+            
+            guard let authenticationSession = authenticationSession, let flows = authenticationSession.flows else {
+                success(false)
+                return
+            }
+            
+            // Return the corresponding VC
+            if self.hasPasswordFlow(inFlows: flows) {
+                success(true)
+            } else {
+                // Flow not supported yet
+                success(false)
+            }
+            
+            }, failure: { (error) in
+                guard let error = error as NSError? else {
```

Comment:
We can also wait for SDK nullability update or add an "unknown error" if the error is nil here. 
```suggestion
                guard let error = error else {
```